### PR TITLE
Porting the MonthCalendar macro

### PR DESCRIPTION
**Pull request by UlrichB22**
This is a first light version of MonthCalendar macro as discussed in #1174 . 

**Documentation**

- see Comments in MonthCalendar.py

- Where should we document macros? User documentation in moin1.9 was maintained in wiki help pages. In moin2 they should be generated from some docs inside the git repo.

**Example**

`<<MonthCalendar(item="WikiCalendar",year=2022,month=6)>>`

- Scripts for migrating from moin1.9 have been added

**Missing features**

- tooltips showing headings of calendar items

- buttons to switch to next or previous month and year (not planned)

**Installation**

- Some translations are added. To activate a recompile is needed:

`python setup.py compile_catalog -l <locale> `

I am glad about your feedback

Best regards,
Ulrich

### CODE REVIEW

File: src/moin/macros/MonthCalendar.py [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+    Calendar of current Page, this year's december:
+    <<MonthCalendar(month=12)>>
+
+    Calendar of December, 2022:
+    <<MonthCalendar(year=2022,month=12)>>
+
+    Calendar of the month two months after December, 2022
+    (maybe doesn't make much sense, but is possible)
+    <<MonthCalendar(year=2022,month=12,offset=+2)>>
```
Comment (RogerHaase) [inline src/moin/macros/MonthCalendar.py:68]: month_offset

File: src/moin/macros/MonthCalendar.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+    Calendar of year 2023 (every month padded to height of 6):
+    ||||||Year 2023||
+    ||<<MonthCalendar(year=2023,month=1,fixed_height=true)>>||<<MonthCalendar(year=2023,month=2,fixed_height=true>>||<<MonthCalendar(year=2023,month=3,fixed_height=true)>>||
+    ||<<MonthCalendar(year=2023,month=4,fixed_height=true)>>||<<MonthCalendar(year=2023,month=5,fixed_height=true>>||<<MonthCalendar(year=2023,month=6,fixed_height=true)>>||
```
Comment (RogerHaase) [inline src/moin/macros/MonthCalendar.py:73]: missing ) in 3 places

File: src/moin/static/css/common.css [hunk 3]
```diff
@@ -689,6 +689,30 @@ ul#moin-rev-navigation { text-align: center; background-color: var(--bg-zebra);
 .moin-rst-optionlist td:first-child,
 .moin-rst-fieldlist td:first-child { font-weight: bold; }
 
+/* MonthCalendar macro */
+
+.calendar a:link, .calendar a:visited { color: #202020; color: var(--cal-text); text-align: center; }
+.calendar table,
+.calendar th,
+.calendar thead,
+.calendar tr { padding: 0; border: 3px solid #FCFCFC; border: 3px solid var(--border-cal); text-align: center; }
```
Comment (RogerHaase) [inline src/moin/static/css/common.css:698]: Sorry, https://github.com/moinwiki/moin/issues/1193 drop support for IE11 was recently closed, so all the duplicate border and color attributes should be removed

border: 3px solid #FCFCFC; border: 3px solid var(--border-cal);

becomes

border: 3px solid var(--border-cal);

**Review by RogerHaase — CHANGES_REQUESTED** [review]

File: src/moin/static/css/common.css (hunk 3 above)
Comment (UlrichB22) [reply src/moin/static/css/common.css]: Many thanks for checking. Your reported issues are fixed.

**Comment by RogerHaase** [comment]
Thanks for helping

**Pull request merged**